app/models/holding.rb

```diff
@@ -27,10 +27,19 @@ def weight
 
   # Basic approximation of cost-basis
   def avg_cost
-    avg_cost = account.entries.trades
-                  .joins("INNER JOIN trades ON trades.id = entries.entryable_id")
-                  .where("trades.security_id = ? AND trades.qty > 0 AND entries.date <= ?", security.id, date)
-                  .average(:price)
+    avg_cost = Trade
```

Comment:
```suggestion
    avg_cost = account.trades
```

Reply:
I'd like to scope this to the account just to stay consistent with our query patterns elsewhere in the app.